cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -42,20 +36,22 @@ pub struct CLedgerInfo {
     pub autobump_ledgers: u32,
 }
 
-impl From<CLedgerInfo> for LedgerInfo {
-    fn from(c: CLedgerInfo) -> Self {
-        let network_passphrase_cstr = unsafe { CStr::from_ptr(c.network_passphrase) };
-        Self {
+impl TryFrom<CLedgerInfo> for LedgerInfo {
+    type Error = anyhow::Error;
+
+    fn try_from(c: CLedgerInfo) -> Result<Self> {
+        let network_passphrase = from_c_string(c.network_passphrase)?;
```

Comment:
Aside: I think we could definitely just calculate the network_id in go and pass that in directly. I don't think preflight every uses the network_passphrase.